snapcraft/common.py

```diff
@@ -58,8 +62,9 @@ def run_output(cmd, **kwargs):
         f.write('\n')
         f.write('exec $*')
         f.flush()
-        return subprocess.check_output(['/bin/sh', f.name] + cmd,
-                                       **kwargs).decode('utf8').strip()
+        return subprocess.check_output([
+            '/bin/sh', f.name] + cmd, **kwargs).decode(
+                sys.getfilesystemencoding()).strip()
```

Comment:
This looks good, but it means that if f.name prints a character that's not ascii when the system doesn't have utf8, it will raise an exception. This was the problem we had in a couple of places. So this possible error might need better handling.